src/Core/Geographic/Coordinates.js

```diff
@@ -0,0 +1,213 @@
+/**
+ * Generated On: 2015-10-5
+ * Class: Coordinates
+ * Description: Coordonnées cartographiques
+ */
+
+import * as THREE from 'three';
+import proj4 from 'proj4';
+import mE from '../Math/MathExtended';
+import Ellipsoid from '../Math/Ellipsoid';
+
+export function ellipsoidSizes() {
+    return {
+        x: 6378137,
+        y: 6356752.3142451793,
+        z: 6378137,
+    };
+}
+
+const ellipsoid = new Ellipsoid(ellipsoidSizes());
+
+export const UNIT = {
+    RADIAN: 0,
+    DEGREE: 1,
+    METER: 2,
+};
+
+function _unitFromProj4Unit(projunit) {
+    if (projunit === 'degrees') {
+        return UNIT.DEGREE;
+    } else if (projunit === 'm') {
+        return UNIT.METER;
+    } else if (projunit === 'radians') {
+        return UNIT.RADIAN;
+    } else {
+        return undefined;
+    }
+}
+
+export function crsToUnit(crs) {
+    switch (crs) {
+        case 'EPSG:4326' : return UNIT.DEGREE;
+        case 'EPSG:4326R': return UNIT.RADIAN;
+        case 'EPSG:4978' : return UNIT.METER;
+        default: {
+            const p = proj4.defs(crs);
+            if (!p) {
+                return undefined;
+            }
+            return _unitFromProj4Unit(p.units);
+        }
+    }
+}
+
+function _crsToUnitWithError(crs) {
+    const u = crsToUnit(crs);
+    if (crs === undefined || u === undefined) {
+        throw new Error(`Invalid crs paramater value '${crs}'`);
+    }
+    return u;
+}
+
+export function assertCrsIsValid(crs) {
+    _crsToUnitWithError(crs);
+}
+
+export function crsIsGeographic(crs) {
+    return (_crsToUnitWithError(crs) != UNIT.METER);
+}
+
+export function crsIsGeocentric(crs) {
+    return (_crsToUnitWithError(crs) == UNIT.METER);
+}
+
+function _assertIsGeographic(crs) {
+    if (!crsIsGeographic(crs)) {
+        throw new Error(`Can't query crs ${crs} long/lat`);
+    }
+}
+
+function _assertIsGeocentric(crs) {
+    if (!crsIsGeocentric(crs)) {
+        throw new Error(`Can't query crs ${crs} x/y/z`);
+    }
+}
+
+// Only support explicit conversions
+function _convert(coordsIn, newCrs) {
+    if (newCrs === coordsIn.crs) {
+        return new Coordinates(newCrs, coordsIn._values);
+    } else {
+        // TODO: of course we don't want to keep this code.
+        // This is mostly a POC until we move to something like proj4.js
+        if (newCrs === 'EPSG:4978') {
+            const cartes = ellipsoid.cartographicToCartesian(coordsIn);
```

Comment:
Err nope actually. `ellipsoid.cartographicToCartesian` uses `longitude(RADIAN)` and `latitude(RADIAN)` so we're good.